### PR TITLE
2.0.0

### PR DESCRIPTION
**Pull request by KiGi**
# Changelog

iOS release v2.0.0 for React Native.

Updates Nami Apple SDK to v2.9.3.

Updates React Native support for bridge to 0.65.1
Updates React Native support for demo apps to 0.66.4

Apple SDK release notes available [here](https://github.com/namiml/nami-apple/releases).

## Breaking Changes

- Changes restore purchases callback to use new form that reports state of restore purchase request instead of success flag.
- UI elements presented by the Nami Apple SDK about the execution of a restore purchase process is now the responsibility of the app using the registerRestorePurchasesHandler callback
- Changes paywall callbacks (linked paywall and paywall meta data for developer ID) to present skus in dictionary as "namiSkus" to reflect NamiPaywall object, was "skus".
- Updated sample apps to reflect new 11.2 minimum iOS requirement (was 11.0).
- Removed all UI modals on Apple that communicate changes related to the purchase process (success and failure) produced by Nami, leaving only Apple system messages

## New Features

- Adds the ability to localize paywalls
- Introduces parameter namiLanguageCode to the NamiConfiguration object to set the language of the paywall you wish to display
- 

## Enhancements

- Improved how the Paywall Creator handles large elements on paywalls to ensure no elements overlap
- Added support for simulators running on M1 processors

##  Bugfixes

- Fixed issue where paywall opened in landscape orientation does not properly render when switched to portrait orientation.
- 

## Maintenance

- Add update script to pod file post-install for iOS to fix compilation errors in react library.
- Bridge code updated React Native to 0.65.1
- Examples apps updated React Native to 0.66.4

### CODE REVIEW

File: examples/Basic/containers/HomeScreen.js [hunk 1]
```diff
@@ -19,6 +19,8 @@ const HomeScreen = (props) => {
   const {navigate} = props.navigation;
   const {purchases} = usePurchasesContext();
 
+  let listenSubscriber;
```
Comment (acantarero) [inline examples/Basic/containers/HomeScreen.js:22]: this is just for the preparePaywall method, right?  if yes we should rename for clarity so they don't think they need to do this for all listeners

File: examples/BasicLinkedPaywall/containers/HomeScreen.js [hunk 2]
```diff
@@ -24,22 +24,30 @@ const HomeScreen = (props) => {
   const {purchases} = usePurchasesContext();
   const {data} = useDataContext();
 
+  let listenSubscriber;
```
Comment (acantarero) [inline examples/BasicLinkedPaywall/containers/HomeScreen.js:27]: if this is only for preparePaywall, let's make sure the name includes preparePaywall so it is clear it only applies to that one listener and not the others

File: ios/RNNami.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -230,6 +230,7 @@
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
 				ENABLE_TESTABILITY = YES;
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
```
Comment (acantarero) [inline ios/RNNami.xcodeproj/project.pbxproj:233]: if we exclude arm64 does that mean the bridge won't work with the simulator?

File: package.json [hunk 4]
```diff
@@ -39,12 +39,13 @@
   ],
   "license": "SEE LICENSE FILE",
   "peerDependencies": {
-    "react": "^16.8.1",
-    "react-native": ">=0.60.0-rc.0 <1.0.x"
+    "react": "^17.0.2",
+    "react-native": "^0.65.1"
```
Comment (acantarero) [inline package.json:43]: let's do `0.65.2` here

File: examples/BasicLinkedPaywall/App.js [hunk 5]
```diff
@@ -55,19 +55,19 @@ const App = () => {
     console.log(NativeModules.NamiMLManagerBridge, 'NamiMLManagerBridge');
```
Comment (acantarero) [inline examples/BasicLinkedPaywall/App.js:55]: we should add a code sample for how to use the new registerRestorePurchasesHandler to the code

Comment (KiGi) [reply examples/BasicLinkedPaywall/App.js:55]: Under the hood the restore purchases call was changed to use the new handler in the bridge, and the results coming back were updated in the LinkedPaywall.js file to look at the state codes returned - we don't have any way at the moment to present these values symbolically though, as there's nothing like "StructsAndEnums" for ReactNative.  Maybe we should translate the numerical values into strings they could look for instead when parsing the dictionary?

File: ios/RNNami.xcodeproj/project.pbxproj (hunk 3 above)
Comment (KiGi) [reply ios/RNNami.xcodeproj/project.pbxproj]: Xcode must have altered this for some reason as I didn't adjust this build setting, removed any exclusion of architectures.

File: examples/Basic/containers/HomeScreen.js (hunk 1 above)
Comment (KiGi) [reply examples/Basic/containers/HomeScreen.js]: Changed to reflect listener is for preparePaywall across all demo apps.

File: examples/BasicLinkedPaywall/components/LinkedPaywall/LinkedPaywall.js [hunk 6]
```diff
@@ -15,42 +15,40 @@ import theme from '../../theme';
...
-    NativeModules.NamiPurchaseManagerBridge.restorePurchases((result) => {
-      console.log('ExampleApp: Nami restorePurchases results was ', result);
-      if (result.success) {
-        NativeModules.NamiPurchaseManagerBridge.purchases((resultInside) => {
-          console.log('ExampleApp: Nami purchases are ', resultInside);
-          console.log('Purchase count is ', resultInside.length);
-          if (resultInside.length > 0) {
+    NativeModules.NamiPurchaseManagerBridge.restorePurchasesWithCompletionHandler((result) => {
+	console.log('Restore Purchases State Change: ', result);
+	console.log('Backgroud image is ', data.paywallMetadata.backgrounds.phone);
```
Comment (acantarero) [inline examples/BasicLinkedPaywall/components/LinkedPaywall/LinkedPaywall.js:24]: this log statement should not be part of this method

Comment (KiGi) [reply examples/BasicLinkedPaywall/components/LinkedPaywall/LinkedPaywall.js]: Removed.

**Review by acantarero — APPROVED** [review]

**Pull request merged**